include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -410,6 +420,10 @@ MultigridPreconditionerBase<dim, Number>::do_initialize_dof_handler_and_constrai
   }
   else // can only be used for triangulations without hanging nodes
   {
+    Assert(tria->has_hanging_nodes() == false,
+           ExcMessage(
+             "Hanging nodes are only supported with the option use_global_coarsening enabled."));
```

Comment:
Independently from this PR, do you think a better name for `use_global_coarsening` would be the name `enable_hanging_nodes` from the perspective of ExaDG users, because the condition in the Assert and also the comment related to the `else` above indicate that this boolean describes something different? Then, I could change this in a subsequent PR.

Reply:
No I don't think so. If it is only about hanging nodes, we would not need any flag and could query the triangulation that information. The flag indicates which transfer implementation should be used. How, we distinguish both implementation is up to discussion; it does not have to be via the bool `use_global_coarsening` but could also be done via an enum.

Reply:
I agree principally, and that's why my suggestion is not `has_hanging_nodes` (which can be answered with yes/no for a given triangulation as you write), but `enable_hanging_nodes`, which reflects that this transfer type can deal with hanging nodes, but also if there are no hanging nodes. Querying the bool parameter from the triangulation would affect the implementation, i.e., it reduces the amount of parameter combinations that are possible. For verification, we found it useful to be able to use both options for the case without hanging nodes.

My point is the following: Currently, the situation might happen that the user writes `use_global_coarsening=true` in the input file, but gets an error message with the suggestion to use a `MultigridType` without h-transfer!? At the same time, the name "global coarsening" insinuates some form of h-transfer in my opinion. This is a contradiction that a general user can not resolve in my opinion. A user of ExaDG does not know that `use_global_coarsening` comes from a MGTransfer implementation in deal.II with a similar name. Even worse, the user can not know that `use_global_coarsening` triggers which implementation to choose for all multigrid types (c, p, h) and not just the h-transfer part of the multigrid algorithm. For the ExaDG user, this boolean is related to enabling hanging nodes in my opinion.

A good documentation of this input parameter would explain the incompatibility of this parameter with p:f:T and refinements (which is not the case currently, see https://github.com/exadg/exadg/blob/965853f349ed30a396b41661ddbf897205de4545/include/exadg/solvers_and_preconditioners/multigrid/multigrid_input_parameters.h#L318). When thinking about how to write this docu so that it is clear to the user, I come to the conclusion that the name `use_global_coarsening` is misleading.

Reply:
OK.